pkg/controller.v1beta1/experiment/manifest/generator_test.go

```diff
@@ -323,22 +308,28 @@ spec:
 				return i
 			}(),
 			ParameterAssignments: newFakeParameterAssignment(),
-			Err:                  true,
-			testDescription:      "Invalid Trial spec in ConfigMap",
+			wantError:            errConvertStringToUnstructuredFailed,
 		},
 	}
 
-	for _, tc := range tcs {
-		actualRunSpec, err := p.GetRunSpecWithHyperParameters(tc.Instance, "trial-name", "trial-namespace", tc.ParameterAssignments)
-		if tc.Err && err == nil {
-			t.Errorf("Case: %v failed. Expected err, got nil", tc.testDescription)
-		} else if !tc.Err {
-			if err != nil {
-				t.Errorf("Case: %v failed. Expected nil, got %v", tc.testDescription, err)
-			} else if !reflect.DeepEqual(expectedRunSpec, actualRunSpec) {
-				t.Errorf("Case: %v failed. Expected %v\n got %v", tc.testDescription, expectedRunSpec.Object, actualRunSpec.Object)
+	testCaseNames := []string{
+		"Run with valid parameters",
+		"Invalid ConfigMap name",
+		"Invalid template path in ConfigMap name",
+		"Invalid trial spec in ConfigMap",
+	}
```

Comment:
@tariq-hasan This indicates that these tests depend on the performing order. But we should not do so.
So, could you resolve the root cause instead of such a hack?

Reply:
Makes sense. I am looking into the issue and will have it resolved.